CK3toEU4/Source/CK3World/Titles/Title.cpp

```diff
@@ -369,3 +368,53 @@ double CK3::Title::getBuildingWeight(const mappers::DevWeightsMapper& devWeights
 								 devWeightsMapper.getDevFromDev() * development;
 	return totalDev;
 }
+
+void CK3::Title::relinkDeFactoVassals()
+{
+	// We're reconstructing defacto hierarchy; we're redirecting our DFvassals from our DFtitle into our DJtitle - if holder of this title holds hold both.
+	// This is important - otherwise secondary titles wouldn't have their own vassals under them but under primary title, breaking PU splitoffs,
+	// and resulting with those secondary duchies/kingdoms being discarded as landless.
+
+	if (name.starts_with("c_") || name.starts_with("b_"))
+		return; // don't bother with counties and below.
+
+	for (const auto& dfVassal: dfVassals)
+	{
+		if (!dfVassal.second)
+			continue;
+
+		if (dfVassal.second->getDFLiege() && dfVassal.second->getDJLiege() && dfVassal.second->getDFLiege()->first != dfVassal.second->getDJLiege()->first)
```

Comment:
For clarity, toss all this logic into a small boolean function named after the intent of the check